### ISSUE TITLE
Bug: ERROR unknown provider: url:https://XX

### ISSUE THREAD

**Issue by WhistleMaster** [qdm12/ddns-updater#203]
**TLDR**:  public IP address with custom http provider gives an error.

1. Is this urgent: _No_
2. DNS provider(s) you use: _N/A_
3. Program version: _Running version latest built on 2021-05-02T18:10:30Z (commit 864a696)_
4. What are you using to run the container: _docker_
5. Extra information (optional)

Name|Value
-- | --
PUBLICIP_FETCHERS | http
PUBLICIP_HTTP_PROVIDERS | https://REDACTED/
PUBLICIPV4_HTTP_PROVIDERS | https://REDACTED/
PUBLICIPV6_HTTP_PROVIDERS | all
PUBLICIP_DNS_PROVIDERS | all

Logs:

```
2021/05/16 07:28:54 INFO Found single setting to update record
2021/05/16 07:28:55 INFO Reading history from database: domain REDACTED host REDACTED
2021/05/16 07:28:55 ERROR unknown provider: url:https://REDACTED/
```

Configuration file (**remove your credentials!**):

```
{
    "settings": [
        {
            "provider": "REDACTED",
            "domain": "REDACTED",
            "host": "REDACTED",
            "provider_ip": true,
            "ip_version": "ipv4",
            "username": "REDACTED",
            "password": "REDACTED"
        }
    ]
}
```

Host OS:
_Linux nuc 4.19.0-16-amd64 #1 SMP Debian 4.19.181-1 (2021-03-19) x86_64 GNU/Linux
Docker version 20.10.6, build 370c289_

**Comment by fredericrous** [comment]
this is the default behavior of ddns-updater. Any suggestion?

**Comment by qdm12** [comment]
It's a bug, sorry I didn't get time to do it. You should be able to specify your own IP url, although depending on the code API this feature may be dropped. I'll get to it soon.

**Comment by WhistleMaster** [comment]
Well indeed, I should be able to 😅 but when I do specify my own IP url, this is the error I get.

I'm not sure to follow, the feature to specify its own IP url is not working at the moment, but will be removed soon ?

**Comment by qdm12** [comment]
Yeah I know, I broke the feature when reworking the public IP fetching https://github.com/qdm12/ddns-updater/commit/8b2e83a69e504e913e8b8627c4e36054e7e8739e by accident. Looking at the code (on mobile) I should be able to restore it, I'll try to do it this weekend.

Out of curiosity, why do you want this feature? I don't exactly recall why I added it in the first place 😄

**Comment by qdm12** [comment]
Actually it was all ready to go, just a small piece of code was missing to check parameters for that custom url option. It should now work 😉

**Comment by WhistleMaster** [comment]
Yeah 🎉 Thanks for the fix ! I'm using it with my own VPS 😉 I'll try it now.
